src/messages/mod.rs

```diff
@@ -43,17 +45,28 @@ pub enum Message {
     Hop(SignedRoutingMessage),
 }
 
+#[derive(Debug, Eq, PartialEq, Hash, Deserialize)]
+#[allow(clippy::large_enum_variant)]
+pub enum MessagePeek {
```

Comment:
Since you accepted the "Partial" naming scheme, maybe use it everywhere for consistency. So this can be `PartialMessage` ?